packages/playground/data-liberation/src/import/WP_Stream_Importer.php

```diff
@@ -241,8 +246,9 @@ private function frontloading_advance_reentrancy_cursor() {
 	 */
 	private function next_frontloading_step() {
 		if ( null === $this->entity_iterator ) {
-			$this->entity_iterator = $this->create_entity_iterator();
-			$this->downloader      = new WP_Attachment_Downloader( $this->options );
+			$this->entity_iterator    = $this->create_entity_iterator();
+			$this->topological_sorter = new WP_Topological_Sorter();
```

Comment:
Ah, so instead of doing a full scan upfront just to sort everything, we do it as we go during the frontloading stage?

Reply:
What does "pretty slow" mean? Perhaps there are XML parsing optimizations we can explore. I want this to support files as large as 300GB.

Reply:
> What does "pretty slow" mean? Perhaps there are XML parsing optimizations we can explore. I want this to support files as large as 300GB.
I tried using an XML with 1 million pages and decided to move to separate steps. It took ~1 minute, so not that slow.

Reply:
1 minute for 1 million pages is excellent, especially considering we haven't done any performance tweaking in the rest of the stack (XML parser, byte streams etc.).

Reply:
More importantly, even if processing is slower than in other WordPress importers, this one can still be faster overall since it has crash recovery baked in.

Reply:
> More importantly, even if processing is slower than in other WordPress importers, this one can still be faster overall since it has crash recovery baked in.
Agreed.